contracts/core/src/contract.rs

```diff
@@ -1078,6 +1114,24 @@ fn execute_bond(
         })?,
         funds: vec![],
     }));
+    let bond_hooks = BOND_HOOKS.load(deps.storage)?;
+    if !bond_hooks.is_empty() {
+        let hook_msg = BondHook {
+            amount: bonded_coin.amount,
+            denom: bonded_coin.denom,
```

Comment:
Is it required to keep tracking of LSM shares as well? Or single amount is enough?

Reply:
The entire BondHook structure is implemented according to this doc: https://www.notion.so/hadron/On-chain-validator-referrals-4b9f3525b47e43a396f43179c71b3466?pvs=4#fa090db1040d4175b42bf07459474423